tapeforms/contrib/foundation.py

```diff
@@ -0,0 +1,70 @@
+from django import forms
+
+from ..mixins import TapeformMixin
+
+
+class FoundationTapeformMixin(TapeformMixin):
+    """
+    Tapeform Mixin to render Foundation compatible forms.
+    (using the template tags provided by `tapeforms`).
+    """
+
+    #: Use a special layout template for Foundation compatible forms.
+    layout_template = 'tapeforms/layouts/foundation.html'
+    #: Use a special field template for Foundation compatible forms.
+    field_template = 'tapeforms/fields/foundation.html'
+    #: Add a special class to invalid field's label.
+    field_label_invalid_css_class = 'is-invalid-label'
+    #: Add a special class to invalid field's widget.
+    widget_invalid_css_class = 'is-invalid-input'
+
+    #: Widgets with multiple inputs require some extra care (don't use ul, etc.)
+    widget_template_overrides = {
+        forms.RadioSelect: 'tapeforms/widgets/foundation_multipleinput.html',
+        forms.CheckboxSelectMultiple: 'tapeforms/widgets/foundation_multipleinput.html'
+    }
+
+    def get_field_label_css_class(self, bound_field):
+        """
+        Appends 'is-invalid-label' if field has errors.
+        """
+        class_name = super().get_field_label_css_class(bound_field)
+
+        if bound_field.errors:
+            if not class_name:
+                class_name = self.field_label_invalid_css_class
+            else:
+                class_name = '{} {}'.format(
+                    class_name, self.field_label_invalid_css_class)
+
+        return class_name
+
+    def get_field_template(self, bound_field, template_name=None):
+        """
+        Uses a special field template for widget with multiple inputs. It only
+        applies if no other template than the default one has been defined.
+        """
+        template_name = super().get_field_template(bound_field, template_name)
+
+        if (template_name == self.field_template and
+                bound_field.field.widget.__class__ in (
```

Comment:
I think we could optimize this further and catch also subclasses of RadioSelect and CheckboxSelectMultiple by replacing the check with something like 
`if template_name == self.field_template and isinstance(bound_field.field.widget, (forms.RadioSelect, forms.CheckboxSelectMultiple)):`

What do you think?

Reply:
It seems better, indeed, I will change that!